src/PlaywrightSharp.Firefox/FirefoxNetworkManager.cs

```diff
@@ -49,19 +49,19 @@ private void OnMessageReceived(object sender, IFirefoxEvent e)
 
         private void OnRequestWillBeSent(NetworkRequestWillBeSentFirefoxEvent e)
         {
-            _requests.TryGetValue(e.RedirectedFrom ?? string.Empty, out var redirectId);
-            var frame = (redirectId != null ? redirectId.Request.Frame : (e.FrameId != null ? _page.FrameManager.Frames[e.FrameId] : null)) as Frame;
+            _requests.TryGetValue(e.RedirectedFrom ?? string.Empty, out var redirected);
+            var frame = redirected?.Request?.Frame ?? (e.FrameId != null ? _page.FrameManager.Frames[e.FrameId] : null);
```

Comment:
I think this is not accurate with pw.

```suggestion
            var frame = redirected != null ? redirected.Request.Frame : (e.FrameId != null ? _page.FrameManager.Frames[e.FrameId] : null);
```